common/src/main/java/com/wynntils/models/character/CharacterStateModel.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © Wynntils 2025.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.character;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Handlers;
+import com.wynntils.core.components.Model;
+import com.wynntils.handlers.bossbar.event.BossBarAddedEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.models.character.bossbar.DeathScreenBar;
+import com.wynntils.models.character.event.CharacterDeathEvent;
+import com.wynntils.models.character.event.CharacterMovedEvent;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.type.Location;
+import java.util.List;
+import net.minecraft.core.Position;
+import net.neoforged.bus.api.SubscribeEvent;
+
+public final class CharacterStateModel extends Model {
```

Comment:
What is the conceptual difference between `Character` and `CharacterState`? I thought just about everything in the Character model was about states..?

Reply:
https://github.com/Wynntils/Wynntils/pull/3202#discussion_r2050171585

This is what I originally thought about it, but the difference is rather subtle right now.

Reply:
I agree that the CharacterModel has always been a bit of a mixed bag, but I can't see a clear way to make the cut as suggested in this PR right now. The extraction of the "AccountModel" was great! It is intuitive and clear. But this...? Nah. Let's say we want to track a new aspect of the character. Would it go into CharacterModel or CharacterStateModel? If there is no clear question you can ask, like "Does it apply to all characters in the user's account?", then we should not make the split, since it will just be arbitrary.

Reply:
I think the distinction here was a bit more clear with the original name. State was supposed to represent physical aspects of the character, like location, death, and in the future, maybe movement prediction, detection of hunted safe zones, in water/lava/submerged states, stuff like that. CharacterModel I intended to keep towards more conceptual aspects of what the character is currently playing. Like id, classtype, level, maybe in the future we will track modifiers like hardcore here as well.

Reply:
So is it like `CharacterEphemeralModel`? Stuff that is valid only for seconds, as opposed to changing e.g. just a few times an hour, or even never changing?

Try to document the distinction between the two models, and get a better name for `CharacterStateModel` and I'm okay with it.